src/connect/components/revision-history/revision/signature.tsx

```diff
@@ -50,9 +30,31 @@ export function Signature(props: SignatureProps) {
     return (
         <Tooltip content={tooltipContent}>
             <span className="flex w-fit cursor-pointer items-center gap-1 rounded-lg border border-gray-200 px-2 py-1">
-                <VerificationStatus />{' '}
-                <Icon name="InfoSquare" className="text-gray-300" size={16} />
+                <VerificationStatus signatures={signatures} />{' '}
+                <Icon className="text-gray-300" name="InfoSquare" size={16} />
             </span>
         </Tooltip>
     );
 }
+
+function VerificationStatus(props: SignatureProps) {
```

Comment:
linter caught that defining this component inline would cause the whole dom tree to be discarded on every render. all needed was to move it out of the component and pass data as props